src/server/cmd/worker/main.go

```diff
@@ -37,6 +37,8 @@ func main() {
 func do(config interface{}) error {
 	// must run InstallJaegerTracer before InitWithKube/pach client initialization
 	tracing.InstallJaegerTracerFromEnv()
+	defer tracing.CloseAndReportTraces()
```

Comment:
I thought `do` didn't exit—does it? Or is this important to error recovery?

Reply:
I'm not sure I understand the question. What do you mean by doesn't exist?

Reply:
Doesn't exit. As in, my understanding is that this defer statement will never run

Reply:
(to clarify, I think `server.Wait()` on l. 100 doesn't return, typically, which means `return server.Wait()` just blocks. Is that not right?)